dowhy/causal_refuters/assess_overlap.py

```diff
@@ -0,0 +1,124 @@
+    Conference on Artificial Intelligence and Statistics (Vol. 108, pp. 788–798). PMLR. https://arxiv.org/abs/1907.04138
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+        Initialize the parameters required for the refuter.
+
+        Arguments are passed through to the `refute_estimate` method. See dowhy.causal_refuters.assess_overlap_overrule
+        for the definition of the `SupportConfig` and `OverlapConfig` dataclasses that define optimization
+        hyperparameters.
+
+        .. warning::
+            This method is only compatible with estimators that use backdoor adjustment, and will attempt to acquire
+            the set of backdoor variables via `self._target_estimand.get_backdoor_variables()`.
+
+        :param: cat_feats: List[str]: List of categorical features, all others will be discretized
+        :param: support_config: SupportConfig: DataClass with configuration options for learning support rules
+        :param: overlap_config: OverlapConfig: DataClass with configuration options for learning overlap rules
+        :param: overlap_eps: float: Defines the range of propensity scores for a point to be considered in the overlap
+            region, with the range defined as `(overlap_eps, 1 - overlap_eps)`, defaults to 0.1
+        :param: overrule_verbose: bool: Enable verbose logging of optimization output, defaults to False
+        :param: support_only: bool: Only fit rules to describe the support region (do not fit overlap rules), defaults to False
+        :param: overlap_only: bool: Only fit rules to describe the overlap region (do not fit support rules), defaults to False
+        """
+        super().__init__(*args, **kwargs)
+        # TODO: Check that the target estimand has backdoor variables?
+        self._backdoor_vars = self._target_estimand.get_backdoor_variables()
+        self._cat_feats = kwargs.pop("cat_feats", [])
+        self._support_config = kwargs.pop("support_config", None)
+        self._overlap_config = kwargs.pop("overlap_config", None)
+        self._overlap_eps = kwargs.pop("overlap_eps", 0.1)
+        if self._overlap_eps < 0 or self._overlap_eps > 1:
+            raise ValueError(f"Value of `overlap_eps` must be in [0, 1], got {self._overlap_eps}")
+        self._support_only = kwargs.pop("support_only", False)
+        self._overlap_only = kwargs.pop("overlap_only", False)
+        self._overrule_verbose = kwargs.pop("overrule_verbose", False)
+
+    def refute_estimate(self, show_progress_bar=False):
+        """
+        Learn overlap and support rules.
+
+        :param show_progress_bar: Not implemented, will raise error if set to True, defaults to False
+        :type show_progress_bar: bool
+        :raises NotImplementedError: Will raise this error if show_progress_bar=True
+        """
+        if show_progress_bar:
+            raise NotImplementedError("No progress bar is available for OverRule")
```

Comment:
Do we want to raise a NotImplementedError here?  Given that the progress bar is secondary functionality, I think we should we just print a warning that it's not implemented but otherwise continue with the rest of the functionality.

Reply:
Changed to a warning, good call